v2/audit/conn/internal/writer/writer.go

```diff
@@ -0,0 +1,63 @@
+// Package writer implements a writer to a remote audit server of any type that can be connected to
+// by a net.Conn object.
+package writer
+
+import (
+	"net"
+	"time"
+
+	"github.com/gostdlib/base/context"
+	"github.com/microsoft/go-otel-audit/v2/audit/msgs"
+)
+
+// ClosePool is a goroutine pool for closing connections to a remote audit server.
+var ClosePool = context.Pool(context.Background()).Sub(context.Background(), "conn.TCPConn.Close")
+
+type now func() time.Time
+
+// Conn is a generic writer to a remote audit server. It can use anything that implements the net.Conn interface.
+type Conn struct {
+	conn net.Conn
+	now  now
+}
+
+// Option is an optional argument for the New() function to configure the connection.
+type Option func(c *Conn) error
```

Comment:
No exported fields on `Conn` makes this of questionable value?

Is this future-proofing?

Reply:
Wouldn't need exported fields for the value, but.....  I'm future proofing an internal/ package which definitely has questionable value.  Removed it.